src/System.Linq.Parallel/tests/Helpers/IntegerRangeSet.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Test
+{
+    // Dummy psuedo-set for verifying we've seen all of a range of integers, and only once.
+    internal class IntegerRangeSet : IEnumerable<KeyValuePair<int, bool>>
+    {
+        private bool[] _seen;
+        private int _start;
+
+        public IntegerRangeSet(int start, int count)
+        {
+            _start = start;
+            _seen = new bool[count];
+        }
+
+        public bool Add(int entry)
+        {
+            Assert.InRange(entry, _start, _start + _seen.Length - 1);
+
+            return _seen[entry - _start] ^= true;
```

Comment:
I was trying to be clever and one-line it, even though I knew this was a potential problem.  Since 99% of the time I was checking the immediate return value anyways (would throw on the second value) I didn't consider it much of a risk...